docker_cli/image.py

```diff
@@ -0,0 +1,430 @@
+import io
+import os
+from shlex import split
+from subprocess import PIPE, STDOUT, CalledProcessError, run
+from sys import stdin
+from typing import (Any, List, Optional, Sequence, Type, TypeVar, Union,
+                    overload)
+
+from .exceptions import CommandNotFoundOnImageError
+
+
+class Image:
+    """A class which holds a reference to a given docker image and provides
+    an interface by which to interact with that image.
```

Comment:
It looks like all of your docstrings begin with a brief description, which is great. Ideally though, I think these initial blurbs should be even more brief than what you have now -- if possible, we should try to limit them to a single line.

Often you'll want to render API docs from your docstrings using a tool like [sphinx](https://www.sphinx-doc.org/en/master/), and it can be useful to create a "summary" table with the short descriptions and a link to the documentation for each documented item. Generally, these brief descriptions shouldn't be more than 1 line of text in order to fit in a summary table like this.

The initial description could be something very short and simple, like this:

```suggestion
    """
    A docker image.
    
    Holds a reference to a given docker image and provides an interface
    by which to interact with that image.
```

Here's some examples of what I mean by summary tables in rendered API docs: [1](https://numpy.org/doc/stable/reference/routines.array-creation.html#from-shape-or-value),[2](https://tophu.readthedocs.io/en/latest/reference.html)